src/test/javascript/spec/service/text-exercise.service.spec.ts

```diff
@@ -100,6 +100,20 @@ describe('TextExercise Service', () => {
             req.flush({ status: 200 });
             expect(requestResult.status).to.equal(200);
         });
+
+        it('should retrieve TextExercise cluster statistics', fakeAsync(() => {
```

Comment:
Do we need a `tick();` here somewhere, or can we remove the `fakeAsync`? I am not a client expert, I just got told that one without the other does not make sense, if I remember correctly. Maybe ask someone who is proficient with that, @Olimbia23, @FrankeLukas?

Reply:
I got tempted to add a `tick();` there as well as I've seen it applied frequently on other tests with `fakeAsync`. However,[ it seemed from here](https://stackoverflow.com/a/56499466) like if there is a `flush()` then a `tick()` might not make much sense. Maybe @Olimbia23 , @FrankeLukas can clear it up if I got it the wrong way.

Reply:
I just double checked it again and it seems that `fakeAsync` is not needed in this case, so @ndricimrr you can remove it.

P.s. in `req.flush({status: 200})` `{status: 200}` is the response body. Hence if you replace it with `{status: 300}` it would also work, which was misleading for me, when running it locally. I would propose to flush `{}` as response in order to avoid confusion.

Reply:
Thanks. Applied the [changes](https://github.com/ls1intum/Artemis/pull/3910/commits/cd4149896ce9bba4b50f529ee34cf9c5298916f1) 👍